core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -269,7 +269,8 @@ private HMSQuery getHMSQuery() {
 
   private <T> T handleFutureResult(Future<T> f) {
     try {
-      return f.get();
+      // Time out if it takes more than 30 seconds
+      return f.get(30, TimeUnit.SECONDS);
```

Comment:
Won't it be best to shutdown "hmsQueriesExecutorService" in case of shutdown, which calls catalog.close();

Reply:
Ok. anyway, that is called. So why was the queries waiting?

Reply:
The query waits for the result on a future. A different thread runs the query and gets the result. The  hive query kept on retrying ( from hive code ) and the future was indefinite wait. It looks like the hive thingy retries for a very very long time...
So thought of timing this out after 30 seconds. Please mind, this is the query on the metastore.